### PR TITLE
rabbitmq: Adding amqps support

### PR DESCRIPTION
**Pull request by joelbax**
Adding support for secure AMQP connections over TLS (amqps).

#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

### CODE REVIEW

**Comment by joelbax** [comment]
@henningw  Sorry to directly bug you but I see that the format check task is failing because a missing space at the end of one line that it's not there in my local file. Do you know how to fix that?

File: src/modules/rabbitmq/rabbitmq.c [hunk 1]
```diff
@@ -557,25 +561,52 @@ static int rabbitmq_connect(amqp_connection_state_t *conn)
 	int ret;
 	int log_ret;
 	//	amqp_rpc_reply_t reply;
+	
+	// amqp_ssl_init_called should only be called once
+	if(amqp_info.ssl && !amqp_ssl_init_called) {
+		amqp_set_initialize_ssl_library(1);
+		amqp_ssl_init_called = 1;
+		LM_DBG("AMQP SSL library inilialized\n");
```
Comment (henningw) [inline src/modules/rabbitmq/rabbitmq.c:569]: Spelling

**Review by henningw — COMMENTED** [review]
Thanks for the PR. I only noticed on small spelling error in the log. Regarding the format check failure, no idea why this happens. Lets wait for eventual other comments and then it can be merged. I can fix eventual remaining errors in the formatting by manually executing the format tool.

**Comment by miconda** [comment]
The PR can be squashed and merged, to get it as a single combined commit, without the branch merge message.

File: src/modules/rabbitmq/rabbitmq.c (hunk 1 above)
Comment (joelbax) [reply src/modules/rabbitmq/rabbitmq.c]: Fixed!

**Comment by linuxmaniac** [comment]
https://docs.github.com/en/get-started/using-git/using-git-rebase-on-the-command-line

**Comment by miconda** [comment]
@joelbax: the message was that the PR is fine for me and if someone considers to merge it should be done with squash option, which appears to the registered developers, not for you. @linuxmaniac's link is good for cli work, but you still can't commit yourself in the mainstream repo.

I just did the merge in that way.

**Pull request merged**